### PR TITLE
run booster catalog verifier as part of CI tests

### PR DESCRIPTION
**Pull request by Ladicek**

### CODE REVIEW

**Comment by gastaldi** [comment]
[test]

File: validate.sh [hunk 1]
```diff
@@ -16,3 +16,8 @@ mvn -q dependency:copy -Dartifact=${GROUPID}:${ARTIFACTID}:${VERSION}:pom -Doutp
 mvn -q dependency:copy -Dartifact=${GROUPID}:${ARTIFACTID}:${VERSION} -DoutputDirectory=${TARGET}
 mvn -q dependency:copy-dependencies -f ${TARGET}/${ARTIFACTID}-${VERSION}.pom -DoutputDirectory=dependencies
 java -cp ${TARGET}/\*:${TARGET}/dependencies/\* ${MAIN} ${ghprbAuthorRepoGitUrl} ${ghprbSourceBranch}
+
+echo "Running https://github.com/rhoar-ci/booster-catalog-verifier.git"
+git clone https://github.com/rhoar-ci/booster-catalog-verifier.git
```
Comment (gastaldi) [inline validate.sh]: You can add `--depth=1` to speed up the clone by not grabbing the whole history

Comment (Ladicek) [reply validate.sh]: Good point, though the history is very small ATM.

**Comment by Ladicek** [comment]
Hmmm... should be doable, if I point each build to its own `-Dmaven.repo.local`...

**Comment by Ladicek** [comment]
No objection to that. I still need to find some time to improve the verifier to run builds in parallel. Hopefully soon, as the WildFly Swarm rush should cease now.

**Comment by Ladicek** [comment]
I made the test run somewhat in parallel (there's a dedicated thread for each runtime, and as part of that, booster builds run sequentially). That takes cca 25 minutes for me locally, which wouldn't be enough for the CentOS CI, but maybe CentOS CI will be able to download the Maven-half-of-the-internet faster.

**Comment by Ladicek** [comment]
[test]

**Comment by gastaldi** [comment]
[test]

**Comment by Ladicek** [comment]
OK, so it's apparently not enough. The builds are IO-bound, was majority of time is taken by downloading Maven artifacts, subsequent booster builds in the same runtime are pretty fast, so not sure if there would be any gain by going fully parallel. Worth a shot, I guess, but I'm not holding my breath.

**Comment by Ladicek** [comment]
I don't know how to make the verification process faster. I was thinking about only checking those boosters that have been changed in the PR, but that's somewhat hard to find (I could probably find the list of files changed in a PR, but how do I then filter the booster catalog?).

I could maybe run the verification on a different Jenkins server, but that's not ideal either. Or would it be possible to extend the build timeout on CentOS CI?

**Comment by quintesse** [comment]
Not sure about the timeout, perhaps @gastaldi knows?

**Comment by gastaldi** [comment]
Created PR in https://github.com/openshiftio/openshiftio-cico-jobs/pull/740 to increase the timeout to 60m

**Comment by gastaldi** [comment]
[test]

**Comment by Ladicek** [comment]
Looks like it worked fine, and built under 30 minutes. Adding more boosters shouldn't increase the time much, because the local Maven repos are reused (i.e., huge initial cost, but subsequent booster builds are in a matter of tens of seconds), so 60 minutes should be safe for quite some time.

How do we proceed? Do we keep the verification in `validate.sh`, or move the validator invocation elsewhere?

**Comment by quintesse** [comment]
personally I'd prefer two different scripts because the whole idea of the `validate.sh` script was that people could run it as a quick test _before_ pushing. I'm not sure they want to run a test on all the boosters, especially if they only changed a single one :)

**Comment by Ladicek** [comment]
Sure, makes sense. I moved the verifier invocation to another script and adjusted `cico_run_tests.sh`.

**Comment by gastaldi** [comment]
[test]

**Comment by Ladicek** [comment]
How do you guys feel about removing the "do not merge" label?

**Comment by gastaldi** [comment]
18 min to build the PR seems a bit too much. Any chance to test only what changed?

**Comment by gastaldi** [comment]
But I'm ok to merge it anyway

**Comment by geoand** [comment]
The functionality introduced by this PR would be an awesome help for all runtimes during releases!
Sure manual review will still be required, but at least we will know that the released boosters aren't broken (which happened to us a few releases back)

**Comment by gastaldi** [comment]
[test]

**Comment by gastaldi** [comment]
Merged. Thank you @Ladicek !

**Pull request merged**